src/request-pipeline/index.js

```diff
@@ -110,6 +112,15 @@ function createReqOpts (ctx) {
     // NOTE: All headers, including 'content-length', are built here.
     var headers = headerTransforms.forRequest(ctx, this);
 
+    // NOTE: Need to remove cookie and authorization headers manually if the request is sent without credentials
+    if (ctx.isXhr && !!ctx.req.headers[XHR_HEADERS.corsSupported] && !ctx.req.headers[XHR_HEADERS.withCredentials]) {
+        delete headers.cookie;
```

Comment:
I think that this logic needs to moves to the `headers-transforms.js` like [this](https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/request-pipeline/header-transforms.js#L49)